File: nim/leap/Iteration1/leap.nim

import math

func isDivisibleBy(date: Natural, denominator: Natural): bool =
  (date mod denominator) == 0

func isLeapYear*(date: Natural): bool =
  isDivisibleBy(date, 4) and not isDivisibleBy(date, 100) or isDivisibleBy(date, 400)

File: cpp/leap/Iteration4/leap.cpp
#include "leap.h"

namespace leap {

	int is_divisible_by(int num, int year)
	{
		return year % num == 0;
	}

	int is_leap_year(int year)
	{
		return is_divisible_by(4, year) && (!is_divisible_by(100, year) || is_divisible_by(400, year));
	}	

}  // namespace leap
